### ISSUE TITLE
[FEATURE REQUEST] Periodic belt tension check by measuring axis length

### ISSUE THREAD

**Issue by murk-sy** [prusa3d/Prusa-Firmware-Buddy#3951]
**Printer type** - Mini, MK4, maybe XL

**Is your feature request related to an existing problem? Please describe.**
With many print hours, the belt tension may be reduced by vibration of the adjustment bolts alone. This can unnoticably but measurably reduce the accuracy of the printer

**Describe the expected functionality**
When axis calibration would be done (which should prompt to check belt tension as well), the printer would measure and store the axis length. Periodically, or at the beginning of every print while the bed is heating, the axis length should be checked. If the measured axis length is longer than first measured by a specific value, it would mean the belt has slack and thus more "length".

**This is a theoretical functionality**
I'm going to be honest, I don't know if this would work, or if it would work reliably - but I feel like this is the best place to suggest it to get some feedback.
Dust, temperature, and a whole bunch of other things *might* affect measurements to the point where this would not make any sense.
This would probably best work if it was just a periodic maintenance that instructs you on the required steps to take, which would include belt tension checking.

**EDIT:** Does not work with current methods (on the Mini at least), see https://github.com/prusa3d/Prusa-Firmware-Buddy/issues/3951#issuecomment-2082123562

**Comment by jpaxer** [comment]
I agree. This would be incredibly useful, especially with input shaping!

**Comment by murk-sy** [comment]
Ran some tests on the Mini and monitoring output with Pronterface.

On one of the axes I have loosened the screws to ~15 Hz lower after the fourth test (ran 7 in total). Meaning, that if it is measurable, there should be a large discrepancy between tests.

Some excel magic later, and:
  | MIN | AVERAGE | MAX | MAX-MIN
-- | -- | -- | -- | --
X speed 1 | -186,57 | -186,57 | -186,57 | 0,00
X speed 2 | -186,87 | -186,82 | -186,71 | 0,16
X speed 3 | -186,88 | -186,81 | -186,40 | 0,48
X speed 4 | -187,04 | -186,40 | -185,91 | 1,13
  | MIN | AVERAGE | MAX | MAX-MIN
Y speed 1 | 183,99 | 183,99 | 184,00 | 0,01
Y speed 2 | 184,01 | 184,08 | 184,17 | 0,16
Y speed 3 | 184,17 | 184,28 | 184,32 | 0,15
Y speed 4 | 184,48 | 184,48 | 184,49 | 0,01

Can you guess which axis was adjusted? That's right, it was the Y axis, which shows essentially no change.

MAX-MIN is essentially useless since I measured some identical values before and after the change:


  | Run 1 | Run 2 | Run 3 | Run 4 | Run 5 | Run 6 | Run 7
-- | -- | -- | -- | -- | -- | -- | --
X speed 1 | -186,57 | -186,57 | -186,57 | -186,57 | -186,57 | -186,57 | -186,57
X speed 2 | -186,86 | -186,71 | -186,86 | -186,87 | -186,87 | -186,73 | -186,87
X speed 3 | -186,88 | -186,88 | -186,40 | -186,88 | -186,88 | -186,88 | -186,88
X speed 4 | -187,04 | -187,04 | -185,91 | -187,04 | -185,92 | -185,91 | -185,91
  | Run 1 | Run 2 | Run 3 | Run 4 | Run 5 | Run 6 | Run 7
Y speed 1 | 183,99 | 183,99 | 184,00 | 183,99 | 184,00 | 183,99 | 184,00
Y speed 2 | 184,17 | 184,01 | 184,01 | 184,17 | 184,01 | 184,01 | 184,17
Y speed 3 | 184,32 | 184,32 | 184,32 | 184,32 | 184,17 | 184,32 | 184,17
Y speed 4 | 184,49 | 184,49 | 184,48 | 184,48 | 184,49 | 184,48 | 184,48

So, not particularly encouraging results.

For now, I will leave it open if a Prusa dev adds their 2 cents, especially if this has been considered before

**Comment by murk-sy** [comment]
Bump
